saleor/order/actions.py

```diff
@@ -216,6 +217,17 @@ def mark_order_as_paid(order: "Order", request_user: "User"):
     payment.captured_amount = order.total.gross.amount
     payment.save(update_fields=["captured_amount", "charge_status", "modified"])
 
+    Transaction.objects.create(
+        payment=payment,
+        action_required=False,
+        kind=TransactionKind.EXTERNAL,
+        token=external_reference or "",
+        is_success=True,
+        amount=order.total.gross.amount,
+        currency=order.total.gross.currency,
+        searchable_key=external_reference or "",
+        gateway_response={},
+    )
```

Comment:
Maybe we should create this only when `external_reference` is provided?

Reply:
@IKarbowiak I was thinking about this, but this will introduce the inconsistency in the database. The order with manual capture but without the reference will not have the transaction, but the order with manual capture and reference will have this.
IMO If we want to record the transaction process we should do that for each case. 
cc @maarcingebala @fowczarek